backend/__tests__/routes.test.ts

```diff
@@ -1,9 +1,27 @@
+import mongoose from 'mongoose';
+import { MongoMemoryServer } from 'mongodb-memory-server';
 import request from 'supertest';
+import { expect } from 'chai';
 import app from '../server';
 
 describe('Unit test example for GET /report', () => {
+  let mongod;
+
+  before(async () => {
+    mongod = await MongoMemoryServer.create();
+    const uri = mongod.getUri();
+    await mongoose.connect(uri);
+  });
+
+  after(async () => {
+    await mongoose.disconnect();
+    await mongod.stop();
+  });
+
   it('should return status 200', async () => {
-    const res = await request(app).get('/report');
-    expect(res.statusCode).toEqual(200);
+    await request(app)
+      .get('/report')
+      .query({ limit: 10, offset: 0 })
+      .expect(200);
```

Comment:
This is good unit test but I'd also suggest adding another integration one for checking format of the output.

Reply:
Done, also I've added sample data with one record for report.